ui/store/actions.js

```diff
@@ -673,6 +673,57 @@ const updateMetamaskStateFromBackground = () => {
   });
 };
 
+export function updateSwapApprovalTransaction(txId, txSwapApproval) {
+  return async (dispatch) => {
+    try {
+      await promisifiedBackground.updateSwapApprovalTransaction(
+        txId,
+        txSwapApproval,
+      );
+    } catch (error) {
+      dispatch(txError(error));
+      dispatch(goHome());
+      log.error(error.message);
+      throw error;
+    }
+
+    try {
+      dispatch(
+        updateTransactionParams(txSwapApproval.id, txSwapApproval.txParams),
+      );
+      const newState = await updateMetamaskStateFromBackground();
+      dispatch(updateMetamaskState(newState));
+      dispatch(showConfTxPage({ id: txSwapApproval.id }));
+      return txSwapApproval;
+    } finally {
+      dispatch(hideLoadingIndication());
+    }
+  };
+}
+
+export function updateSwapTransaction(txId, txSwap) {
+  return async (dispatch) => {
+    try {
+      await promisifiedBackground.updateSwapTransaction(txId, txSwap);
+    } catch (error) {
+      dispatch(txError(error));
+      dispatch(goHome());
+      log.error(error.message);
+      throw error;
+    }
+
+    try {
+      dispatch(updateTransactionParams(txSwap.id, txSwap.txParams));
+      const newState = await updateMetamaskStateFromBackground();
+      dispatch(updateMetamaskState(newState));
+      dispatch(showConfTxPage({ id: txSwap.id }));
+      return txSwap;
+    } finally {
+      dispatch(hideLoadingIndication());
```

Comment:
I believe we can remove these `hideLoadingIndication` actions since we removed `showLoadingIndication`. 

Same comment for `hideLoadingIndication` in `updateSwapApprovalTransaction`

Reply:
Fixes [here](https://github.com/MetaMask/metamask-extension/pull/13641/commits/b96851034b9e1b43000aac2c4db71efca309ec08)